packages/react-variant-view/src/component/pathogenicity/ClinVarInterpretation.tsx

```diff
@@ -0,0 +1,105 @@
+import _ from 'lodash';
+import * as React from 'react';
+
+import { observer } from 'mobx-react';
+import featureTableStyle from '../featureTable/FeatureTable.module.scss';
+import { computed } from 'mobx';
+import { ClinVar } from 'genome-nexus-ts-api-client';
+import { Button } from 'react-bootstrap';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+
+interface IClinVarInterpretationProps {
+    clinVar: ClinVar | undefined;
+}
+
+enum ClinVarOrigin {
+    GERMLINE = 'germline',
+    SOMATIC = 'somatic',
+}
+
+@observer
+class ClinVarInterpretation extends React.Component<
+    IClinVarInterpretationProps
+> {
+    @computed get rcvCountMap() {
+        if (this.props.clinVar) {
+            const clinVar = this.props.clinVar;
+            const filteredRcv = clinVar.rcv.filter(
+                rcv =>
+                    rcv.origin === ClinVarOrigin.GERMLINE ||
+                    rcv.origin === ClinVarOrigin.SOMATIC
+            );
+            const rcvMap = _.groupBy(filteredRcv, d => d.origin);
+            const rcvCountMap = _.mapValues(rcvMap, rcvs =>
+                _.countBy(rcvs, rcv => rcv.clinicalSignificance)
+            );
+            return rcvCountMap;
+        } else {
+            return undefined;
+        }
+    }
+
+    @computed get clinVarContent() {
```

Comment:
This function is a bit difficult to follow. Seems like here we are doing both data transformation and rendering at the same time. It might be better to have a separate method to transform the data into the structure that we need, and a simpler render function consuming the transformed data.